File: benchmark/bench/strings.js
// build __N__ small strings, join them, then rolling-hash the char codes.
function main() {
  var N = __N__;
  var parts = [];
  for (var i = 0; i < N; i++) {
    parts.push("item" + (i % 1000));
  }
  var s = parts.join(",");
  var sum = 0;
  for (var i = 0; i < s.length; i++) {
    sum = (sum * 31 + s.charCodeAt(i)) % 1000000007;
  }
  return sum;
}
var result = "" + main();
result;

File: benchmark/bench/strings.lua
-- build __N__ small strings, join them, then rolling-hash the char codes.
local function main()
  local N = __N__
  local parts = {}
  for i = 0, N - 1 do
    parts[#parts + 1] = "item" .. (i % 1000)
  end
  local s = table.concat(parts, ",")
  local sum = 0
  for i = 1, #s do
    sum = (sum * 31 + string.byte(s, i)) % 1000000007
  end
  return sum
end
result = string.format("%d", main())
